src/modules/PowerOCR/PowerOCR/Helpers/ImageMethods.cs

```diff
@@ -91,35 +112,17 @@ internal static async Task<string> GetClickedWord(Window passedWindow, System.Wi
 
     public static async Task<string> ExtractText(Bitmap bmp, Language? preferredLanguage, System.Windows.Point? singlePoint = null)
     {
-        Language? selectedLanguage = preferredLanguage;
-        if (selectedLanguage == null)
-        {
-            selectedLanguage = GetOCRLanguage();
-        }
-
+        Language? selectedLanguage = preferredLanguage ?? GetOCRLanguage();
         if (selectedLanguage == null)
         {
             return string.Empty;
         }
 
-        bool isCJKLang = false;
-
-        if (selectedLanguage.LanguageTag.StartsWith("zh", StringComparison.InvariantCultureIgnoreCase) == true)
-        {
-            isCJKLang = true;
-        }
-        else if (selectedLanguage.LanguageTag.StartsWith("ja", StringComparison.InvariantCultureIgnoreCase) == true)
-        {
-            isCJKLang = true;
-        }
-        else if (selectedLanguage.LanguageTag.StartsWith("ko", StringComparison.InvariantCultureIgnoreCase) == true)
-        {
-            isCJKLang = true;
-        }
-
         XmlLanguage lang = XmlLanguage.GetLanguage(selectedLanguage.LanguageTag);
         CultureInfo culture = lang.GetEquivalentCulture();
 
+        bool isSpaceJoiningLang = LanguageHelper.IsLanguageSpaceJoining(selectedLanguage);
```

Comment:
Am I getting this wrong, or `LanguageHelper.IsLanguageSpaceJoining` reverts previous logic?
Before:
`if "zh" isCJKLang = true`
Now:
`if "zh" isSpaceJoiningLang = false`

Reply:
Good catch! The logic is reversed and passed into `OcrExtensions.GetTextFromOcrLine` it was used correctly. It was also used on line 181, but there are no languages which are non-space joining and right to left, so I just removed that line.